### PR TITLE
Info parameter not returned for GENERIC_ERROR resultcode

### PR DESCRIPTION
**Pull request by pold500**
Info parameter not returned for GENERIC_ERROR resultcode for PerformAudioPassThru when watchdog timeout from HMI expires.

Changed:
* src/components/application_manager/include/application_manager/commands/command_request_impl.h
* src/components/application_manager/src/commands/command_request_impl.cc
  Added 'info' parameter into 'onTimeOut' virtual method
* src/components/application_manager/src/commands/mobile/perform_audio_pass_thru_request.cc
  Added info parameter if request is out of time
Changed virtual method signature to match descendants

@LuxoftAKutsan 
@tkireva 
@okozlovlux

### CODE REVIEW

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h [hunk 1]
```diff
@@ -114,7 +114,7 @@ class CommandRequestImpl : public CommandImpl,
    * has exceed it's limit
    *
    */
-  virtual void onTimeOut();
+  virtual void onTimeOut(const std::string &info = "");
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @pold500 virtual function with default parameters is very very bad. It even pointed out in google coding style

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 2]
```diff
@@ -224,9 +224,8 @@ void CommandRequestImpl::onTimeOut() {
                                             function_id(),
                                             correlation_id(),
                                             mobile_api::Result::GENERIC_ERROR);
-
+  (*response)[strings::msg_params][strings::info] = info;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @pold500 with that approach SDL will send empty info parameter for each timed out request

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 3]
```diff
@@ -224,9 +224,8 @@ void CommandRequestImpl::onTimeOut() {
...
   AddTimeOutComponentInfoToMessage(response);
```
Comment (dcherniev) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @pold500 Have you checked this method? Purpose of this method is exactly to create the "info" param for onTimeOut response, and currently if this method will run successfully, your received info parameter will be simply replaced inside of this method. So technically either you should update expected info inside of this method, or maybe there is something wrong with the script that it does not pass trough this method for some reason.

File: src/components/application_manager/src/commands/mobile/perform_audio_pass_thru_request.cc [hunk 4]
```diff
@@ -62,7 +62,8 @@ void PerformAudioPassThruRequest::onTimeOut() {
   LOG4CXX_AUTO_TRACE(logger_);
 
   FinishTTSSpeak();
-  CommandRequestImpl::onTimeOut();
+  const std::string info = awaiting_ui_response_ ? "UI component does not respond":"";
```
Comment (dcherniev) [inline src/components/application_manager/src/commands/mobile/perform_audio_pass_thru_request.cc]: @pold500 I'm a bit confused, is it possible to get inside of this onTimeOut function having false "awaiting_ui_response_" parameter?

File: src/components/application_manager/src/commands/command_request_impl.cc (hunk 2 above)
Comment (dcherniev) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @pold500 Yes, this way the "info" parameter will always be present in this response which may be an error behavior for some cases.

File: src/components/application_manager/src/commands/command_request_impl.cc (hunk 3 above)
Comment (pold500) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @dcherniev eventually, it does go there, but there's no "info" field afterwards, so I had to add it manually.

File: src/components/application_manager/src/commands/mobile/perform_audio_pass_thru_request.cc (hunk 4 above)
Comment (pold500) [inline src/components/application_manager/src/commands/mobile/perform_audio_pass_thru_request.cc]: @dcherniev I don't know, is there's a way to check it?

File: src/components/application_manager/src/commands/command_request_impl.cc (hunk 3 above)
Comment (dcherniev) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @pold500 Well I guess that means that the test hits 1 of the return false conditions in this method, have you checked what condition was it?

Comment (tkireva) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @pold500 Recently I was working on similar issue with missing info parameter in message with return code GENERIC_ERROR. It turned out that in this test script one of all messages that are sent to the SDL is "broken" and the reason of returning GENERIC_ERROR is different from the expected in the test script and as a result the info parameter also wasn't with the expected value. So, I advise you to get sure about the reason of returning GENERIC_ERROR in your case, it may be different from the expectations in the script.

File: src/components/application_manager/src/commands/mobile/perform_audio_pass_thru_request.cc (hunk 4 above)
Comment (tkireva) [inline src/components/application_manager/src/commands/mobile/perform_audio_pass_thru_request.cc]: @pold500 @dcherniev 
This function is called in case of UI timed out or TTS timed out. It is common for both interfaces and at this point of view it is possible awaiting_ui_response to be false inside. This may be verified by some test script of this CRQ - for missing UI part and missing TTS part.

**Comment by pold500** [comment]
@dcherniev please review
@LuxoftAKutsan please review
@istoilovgithub please review

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 5]
```diff
@@ -799,68 +821,31 @@ void CommandRequestImpl::AddTimeOutComponentInfoToMessage(
...
+         m_notRespondingInterfaces.push_back(static_cast<HmiInterfaces::InterfaceID>(interface_id));
+      }
   }
+  std::string not_responding_components;
+  std::for_each(std::begin(m_notRespondingInterfaces), std::end(m_notRespondingInterfaces), [&m_notRespondingInterfaces,
+                &not_responding_components]
+                (const HmiInterfaces::InterfaceID& not_responding_interface) {
+      const auto component_name = GetComponentNameFromInterface(not_responding_interface);
+      not_responding_components+= component_name;
+      if(not_responding_interface != *(std::end(m_notRespondingInterfaces) - 1)) {
```
Comment (pold500) [inline src/components/application_manager/src/commands/command_request_impl.cc]: I think I've stopped a mistake there. Needs to be fixed.

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 6]
```diff
@@ -790,6 +788,29 @@ bool CommandRequestImpl::IsResultCodeUnsupported(
...
+    case HmiInterfaces::HMI_INTERFACE_VehicleInfo:
+        return "VehicleInfo";
+    case HmiInterfaces::HMI_INTERFACE_SDL:
+        return "SDL";
+    default:
+        return "Unknown type";
+    }
+}
+
 void CommandRequestImpl::AddTimeOutComponentInfoToMessage(
```
Comment (istoilovgithub) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @pold500 IMO using request id returned by the member Command::function_id you can find exact HMI type that was timed out. Something like this:

HmiInterfaces& hmi_interfaces = application_manager_.hmi_interfaces();
HmiInterfaces::InterfaceID interface =  hmi_interfaces.GetInterfaceFromFunction(function_id());
const std::string& component_name = GetComponentNameFromInterface(interface);
...

**Comment by istoilovgithub** [comment]
@pold500 One more general comments from my side: Please rephrase your commits to follow guidelines. Squash your commits when possible. Prefer rebasing vs merging to resolve conflicts, it keeps commit history simpler.

**Comment by pold500** [comment]
Please review @LuxoftAKutsan @istoilovgithub

**Comment by VVeremjova** [comment]
@pold500 You wrote in description that there was changed command_request_impl.h, which actually doesn't

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 7]
```diff
@@ -790,6 +788,29 @@ bool CommandRequestImpl::IsResultCodeUnsupported(
          (first.is_unsupported_resource && second.is_unsupported_resource);
 }
 
+std::string GetComponentNameFromInterface(const HmiInterfaces::InterfaceID& interface) {
```
Comment (VVeremjova) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @pold500 I have checked locally your code with scripts 027_ATF_P_PerformAudioPassThru_UI_timeout_TTS_success.lua
028_ATF_P_PerformAudioPassThru_TTS_timeout_UI_success.lua
and they don't pass. The error is:
expected: UI component does not respond, actual value: Buttons, BasicCommunication, SDL component does not respond
The same situation with any script for TTS, VR not respond. 
Please fix logic and check locally with scripts for reproducing defect

Comment (pold500) [inline src/components/application_manager/src/commands/command_request_impl.cc]: yes I know about that but that's not logic which is wrong it's script.

**Comment by pold500** [comment]
@VVeremjova please review
@LuxoftAKutsan please review

File: src/components/application_manager/test/commands/mobile/alert_request_test.cc [hunk 8]
```diff
@@ -235,6 +235,8 @@ TEST_F(AlertRequestTest, OnTimeout_GENERIC_ERROR) {
       (*ui_command_result)[am::strings::msg_params][am::strings::result_code]
           .asInt(),
       static_cast<int32_t>(am::mobile_api::Result::GENERIC_ERROR));
+  EXPECT_EQ((*ui_command_result)[am::strings::msg_params][am::strings::info].asString(),
```
Comment (VVeremjova) [inline src/components/application_manager/test/commands/mobile/alert_request_test.cc]: @pold500 It looks incorrect - according requirements info should not be empty if there is GENERIC_ERROR after timeout expired. Please correct this check

File: src/components/application_manager/test/commands/command_request_impl_test.cc [hunk 9]
```diff
@@ -185,6 +185,7 @@ TEST_F(CommandRequestImplTest, OnTimeOut_StateAwaitingHMIResponse_SUCCESS) {
 
   command->onTimeOut();
 
+  EXPECT_TRUE((*dummy_msg)[am::strings::msg_params][am::strings::info].asString().empty());
```
Comment (VVeremjova) [inline src/components/application_manager/test/commands/command_request_impl_test.cc]: @pold500 Please add some comment why here info should be empty or change this check

File: src/components/application_manager/test/commands/mobile/perform_audio_pass_thru_test.cc [hunk 10]
```diff
@@ -346,7 +346,7 @@ TEST_F(PerformAudioPassThruRequestTest, OnTimeout_GENERIC_ERROR) {
   command->onTimeOut();
 
   ResultCommandExpectations(
-      msg_mobile_response, NULL, am::mobile_api::Result::GENERIC_ERROR, false);
+      msg_mobile_response, "", am::mobile_api::Result::GENERIC_ERROR, false);
```
Comment (VVeremjova) [inline src/components/application_manager/test/commands/mobile/perform_audio_pass_thru_test.cc]: @pold500 Again, after timeout info shouldn't be empty

File: src/components/application_manager/test/commands/mobile/perform_interaction_test.cc [hunk 11]
```diff
@@ -236,6 +237,7 @@ TEST_F(PerformInteractionRequestTest, OnTimeout_VR_GENERIC_ERROR) {
   EXPECT_EQ(
       (*vr_command_result)[strings::msg_params][strings::result_code].asInt(),
       static_cast<int32_t>(am::mobile_api::Result::GENERIC_ERROR));
+  EXPECT_EQ((*vr_command_result)[strings::msg_params][strings::info].asString(), "");
```
Comment (VVeremjova) [inline src/components/application_manager/test/commands/mobile/perform_interaction_test.cc]: @pold500 According name of test info should be "VR component does not respond"

File: src/components/application_manager/test/commands/mobile/set_audio_streaming_indicator_test.cc [hunk 12]
```diff
@@ -494,18 +494,14 @@ TEST_F(SetAudioStreamingIndicatorRequestTest,
...
   ApplicationSharedPtr mock_app_empty;
   EXPECT_CALL(app_mngr_, application(_))
-      .Times(2)
+      .Times(1)
       .WillRepeatedly(Return(mock_app_empty));
 
   EXPECT_CALL(*mock_app_, RemoveIndicatorWaitForResponse(_)).Times(0);
   EXPECT_CALL(
       mock_message_helper_,
       CreateNegativeResponse(_, _, _, am::mobile_api::Result::GENERIC_ERROR))
```
Comment (VVeremjova) [inline src/components/application_manager/test/commands/mobile/set_audio_streaming_indicator_test.cc]: @pold500 It doesn't check info

**Comment by dcherniev** [comment]
@pold500 Your pull request had become quite complicated, so it is hard to review it right now, thus I have some generall questions:
Can you squash your changes to few logical blocks for example "changes related to defect fix" and "changes related to unit tests update"?
Please update your pull request description - it's the very first comment in this pull request, and write what changes in what files were made and what purpose each change has in scope of your pull request.
As Ivo has pointed out: please rebase your pull request on latest version of extended_policy_functionality, this will clear your pull request from changes that was not made by you.
If you have changes that are no longer valid and erased in one of your future commits please squash them so they don't uppear in diff history since it is really confusing to read your changes "step by step". This  also applyes to minor fixes like formating code stile fix, it can be squashed in previous commit since there is no point to review changes related to formatting.
Also please be aware that it is a rule to add an answer to each comment that you recieved during the review. It should be a comment with your opinion if you do not aggree, or comment with link to changes that were made according to reviewer request. Do not leave unanswered comments since your reviewer will be unaware that you have completed requested change. Also this will help you remember on what comments you have answered, and what comments still should be checked.
So overall, please update your pull request to make it clear for your reviewers, thank you!

File: src/components/application_manager/test/commands/mobile/set_audio_streaming_indicator_test.cc (hunk 12 above)
Comment (pold500) [inline src/components/application_manager/test/commands/mobile/set_audio_streaming_indicator_test.cc]: @VVeremjova fixed in commit https://github.com/smartdevicelink/sdl_core/pull/1368/commits/e1e998ffc82c12f956a5d10777d9061aa13daae2

File: src/components/application_manager/test/commands/mobile/alert_request_test.cc (hunk 8 above)
Comment (pold500) [inline src/components/application_manager/test/commands/mobile/alert_request_test.cc]: @VVeremjova fixed in https://github.com/smartdevicelink/sdl_core/pull/1368/commits/e1e998ffc82c12f956a5d10777d9061aa13daae2

File: src/components/application_manager/test/commands/command_request_impl_test.cc (hunk 9 above)
Comment (pold500) [inline src/components/application_manager/test/commands/command_request_impl_test.cc]: @VVeremjova fixed in https://github.com/smartdevicelink/sdl_core/pull/1368/commits/e1e998ffc82c12f956a5d10777d9061aa13daae2

File: src/components/application_manager/test/commands/mobile/perform_audio_pass_thru_test.cc (hunk 10 above)
Comment (pold500) [inline src/components/application_manager/test/commands/mobile/perform_audio_pass_thru_test.cc]: @VVeremjova fixed in https://github.com/smartdevicelink/sdl_core/pull/1368/commits/e1e998ffc82c12f956a5d10777d9061aa13daae2

File: src/components/application_manager/test/commands/mobile/perform_interaction_test.cc (hunk 11 above)
Comment (pold500) [inline src/components/application_manager/test/commands/mobile/perform_interaction_test.cc]: @VVeremjova fixed in https://github.com/smartdevicelink/sdl_core/pull/1368/commits/e1e998ffc82c12f956a5d10777d9061aa13daae2

File: src/components/application_manager/test/commands/mobile/set_audio_streaming_indicator_test.cc (hunk 12 above)
Comment (pold500) [inline src/components/application_manager/test/commands/mobile/set_audio_streaming_indicator_test.cc]: @VVeremjova fixed in https://github.com/smartdevicelink/sdl_core/pull/1368/commits/e1e998ffc82c12f956a5d10777d9061aa13daae2

**Comment by pold500** [comment]
@istoilovgithub squashed commits to form much cleaner and simpler representation

File: src/components/application_manager/src/commands/command_request_impl.cc (hunk 6 above)
Comment (istoilovgithub) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @pold500 I am still thinking that this is the right way to fix the issue, did you give it a try?

Comment (pold500) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @istoilovgithub yeah I tried that, unfortunately, it didn't work.

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 13]
```diff
@@ -225,13 +225,12 @@ void CommandRequestImpl::onTimeOut() {
...
-
-  AddTimeOutComponentInfoToMessage(response);
-
+  AddTimeOutComponentInfoToMessage(*response, awaiting_response_interfaces_);
   application_manager_.ManageMobileCommand(response, ORIGIN_SDL);
 }
 
-void CommandRequestImpl::on_event(const event_engine::Event& event) {}
+void CommandRequestImpl::on_event(const event_engine::Event& event) {
+}
```
Comment (dcherniev) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @pold500 I think that this change can be dropped.

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 14]
```diff
@@ -248,9 +247,11 @@ void CommandRequestImpl::SendResponse(
     current_state_ = kCompleted;
   }
 
-  smart_objects::SmartObjectSPtr result =
-      utils::MakeShared<smart_objects::SmartObject>();
-
+  smart_objects::SmartObjectSPtr result = new smart_objects::SmartObject;
+  if (!result) {
+    LOG4CXX_ERROR(logger_, "Memory allocation failed.");
```
Comment (dcherniev) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @pold500 Please do not use such allocation checks, it is impossible to reach it by unit tests. Please use 'utils::MakeShared' to allocete your SmartObject instead of 'new' operator.

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 15]
```diff
@@ -789,75 +811,69 @@ bool CommandRequestImpl::IsResultCodeUnsupported(
...
-    case mobile_apis::AppHMIType::BACKGROUND_PROCESS:
-      return "Background Process";
-    case mobile_apis::AppHMIType::TESTING:
-      return "Testing";
-    case mobile_apis::AppHMIType::SYSTEM:
-      return "System";
-    default:
-      return "Invalid Component Type";
+  std::vector<HmiInterfaces::InterfaceID> not_responding_interfaces_vector;
+  // we copy contents to vector so we can have iterator that will allow us to do
```
Comment (dcherniev) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @pold500 Please add some spaces here, clarify in your comment what vector is considered here. Is it related to vector above? In that case shouldn't the description be before it?

File: src/components/application_manager/test/commands/command_request_impl_test.cc [hunk 16]
```diff
@@ -164,7 +164,7 @@ TEST_F(CommandRequestImplTest, OnTimeOut_StateCompleted_UNSUCCESS) {
   EXPECT_EQ(RequestState::kCompleted, command->current_state());
 }
 
-TEST_F(CommandRequestImplTest, OnTimeOut_StateAwaitingHMIResponse_SUCCESS) {
+TEST_F(CommandRequestImplTest, DISABLED_OnTimeOut_StateAwaitingHMIResponse_SUCCESS) {
```
Comment (dcherniev) [inline src/components/application_manager/test/commands/command_request_impl_test.cc]: @pold500 You have disabled 6 unit tests, please add a description why each of the tests was disabled by either adding a TODO or creating a common description in commit message.

File: src/components/application_manager/test/commands/mobile/set_audio_streaming_indicator_test.cc [hunk 17]
```diff
@@ -494,18 +500,14 @@ TEST_F(SetAudioStreamingIndicatorRequestTest,
...
       .WillOnce(Return(msg_mobile_response));
-  EXPECT_CALL(
-      app_mngr_,
-      ManageMobileCommand(_, am::commands::Command::CommandOrigin::ORIGIN_SDL))
-      .WillOnce(DoAll(SaveArg<0>(&msg_mobile_response), Return(true)));
 
   command->onTimeOut();
```
Comment (dcherniev) [inline src/components/application_manager/test/commands/mobile/set_audio_streaming_indicator_test.cc]: @pold500 Title of this commit is too big, please reword it to make it more compact and instead extend the commit description.

**Comment by Itileda** [comment]
Hi @LuxoftAKutsan , @istoilovgithub , @tkireva , @VVeremjova , @dcherniev 

Pull request is updated by Pavel. Please review it ASAP.

**Comment by tkireva** [comment]
@pold500 Have you tested this situation for timed out RPC for all(most of) the RPCs. It is common requirement for all of them to send the name of the component. I ask you this, because I see you use method StartAwaitForInterface() only in PerformAudioPassThruRequest and ReadDIDRequest. I also think that Ivo's suggestion is really appropriate. What was the problem with it? Do you need some help?

**Comment by pold500** [comment]
@tkireva I know this but I won't update update all rpc's in one pull request. If there will be corresponding defects, I suggest assign them to me and I fix that rpc's.
About Ivo's solution,it didn't work. Using function_id did not help.

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 18]
```diff
@@ -789,75 +808,73 @@ bool CommandRequestImpl::IsResultCodeUnsupported(
...
-      return "Background Process";
-    case mobile_apis::AppHMIType::TESTING:
-      return "Testing";
-    case mobile_apis::AppHMIType::SYSTEM:
-      return "System";
-    default:
-      return "Invalid Component Type";
+  // We copy contents to vector so we can have iterator that will allow us to do
+  // operator-() on it, it will be used in the check later
+  std::vector<HmiInterfaces::InterfaceID> not_responding_interfaces_vector;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @pold500 print is this way.
http://codepad.org/0G4eB69b

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 19]
```diff
@@ -780,6 +778,27 @@ const CommandParametersPermissions& CommandRequestImpl::parameters_permissions()
...
+  std::set<HmiInterfaces::InterfaceID>::const_iterator it =
+      awaiting_response_interfaces_.find(interface_id);
+  return (it != awaiting_response_interfaces_.end());
+}
+
+void CommandRequestImpl::EndAwaitForInterface(
+    const HmiInterfaces::InterfaceID& interface_id) {
+  std::set<HmiInterfaces::InterfaceID>::const_iterator it =
+      awaiting_response_interfaces_.find(interface_id);
+  if (it != awaiting_response_interfaces_.end()) {
```
Comment (pold500) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @LuxoftAKutsan you mean that I should check if StartAwawit was called, right?

File: src/components/application_manager/src/commands/command_request_impl.cc (hunk 6 above)
Comment (istoilovgithub) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @pold500 My intention is to find as much generic solution as possible, since the current one is command specific. Anyway, if there is no reliable way to find the HMI interface of the timeout command, I am fine with your changes, but what about the other commands, since the same requirement is also applicable to them?

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h [hunk 20]
```diff
@@ -268,6 +268,30 @@ class CommandRequestImpl : public CommandImpl,
    */
   const CommandParametersPermissions& parameters_permissions() const;
 
+  /**
+   * @brief This set stores all the interfaces which are awaited by SDL to
+            return a response on some request
+  */
+  std::set<HmiInterfaces::InterfaceID> awaiting_response_interfaces_;
```
Comment (VVeremjova) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @pold500 Please move it below functions' declarations

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h [hunk 21]
```diff
@@ -296,19 +320,22 @@ class CommandRequestImpl : public CommandImpl,
...
    * @brief Method transforms AppHMIType to string
    * @param enum AppHMIType app_hmi_type contains enum value
    * @return string of AppHMIType
    */
   std::string AppHMITypeToString(
       const mobile_apis::AppHMIType::eType app_hmi_type) const;
+
+  /**
+   * @brief Adds information to result message "info" param in case of GENERIC_ERROR
+            about component which not repsonded in time
```
Comment (VVeremjova) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @pold500 typo in responded

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 22]
```diff
@@ -789,75 +808,73 @@ bool CommandRequestImpl::IsResultCodeUnsupported(
...
-      app_type->getType()) {
-    LOG4CXX_WARN(logger_, "Application types are not an array!");
-    return;
-  }
-
-  const SmartObject& app_type_so = app_type->getElement(0);
-  if (NsSmartDeviceLink::NsSmartObjects::invalid_object_value == app_type_so) {
-    LOG4CXX_WARN(logger_, "Invalid object for component type!");
+  if (not_responding_interfaces.empty()) {
+    LOG4CXX_DEBUG(logger_,
```
Comment (VVeremjova) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @pold500 Probably it should be Warn

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 23]
```diff
@@ -789,75 +808,73 @@ bool CommandRequestImpl::IsResultCodeUnsupported(
...
+    const std::string component_name =
+        GetComponentNameFromInterface(not_responding_interface);
+    not_responding_interfaces_string += component_name;
+    if (it != (std::end(not_responding_interfaces_vector) - 1)) {
+      not_responding_interfaces_string += ", ";
+    }
+  }
+  LOG4CXX_DEBUG(
+      logger_,
+      "not_responding_interfaces_string: " << not_responding_interfaces_string);
```
Comment (VVeremjova) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @pold500 Change message for more readable: e.g. `"Not respond : "<< not_responding_interfaces_string` or print the same string as component_info

File: src/components/application_manager/src/commands/command_request_impl.cc (hunk 18 above)
Comment (pold500) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @LuxoftAKutsan about solution with "accumulate": template type of set is enum, not a string, so I still will have to convert set to set of strings or vector of strings for your solution to work.

Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/command_request_impl.cc]:
```
#include <set>
#include <iostream>
#include <string>
#include <numeric>

enum eType {
  X,
  Y
};

std::string convert(eType t) {
  return (X == t) ?  "X": "Y";
}

std::string InfoInterfaceSeparator(const std::string& curr, const eType& val) {
    return curr.empty() ? convert(val) : curr + ", " + convert(val);
}

int main() {
  std::set<eType> myset;
  myset.insert(X);
  std::string res = std::accumulate(
      myset.begin(),
      myset.end(),
      std::string(""),
      InfoInterfaceSeparator
      );
  std::cout << res << std::endl;
}

~                                                                                                                                                                                                                                             
~                                                                                                                                                                                                                                             
~                                                                                                                                                                                                                                             
~     
```

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h (hunk 21 above)
Comment (pold500) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @VVeremjova all fixed in https://github.com/smartdevicelink/sdl_core/pull/1368/commits/6cb301a444f95a33a443f5c0438a182f3b6c9e16

File: src/components/application_manager/src/commands/command_request_impl.cc (hunk 19 above)
Comment (pold500) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @LuxoftAKutsan fixed in commit https://github.com/smartdevicelink/sdl_core/pull/1368/commits/6cb301a444f95a33a443f5c0438a182f3b6c9e16

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h (hunk 21 above)
Comment (VVeremjova) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @pold500 'repsonded' - > 'responded'

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 24]
```diff
@@ -105,7 +105,7 @@ const std::string CreateInfoForUnsupportedResult(
 
 bool CheckResultCode(const ResponseInfo& first, const ResponseInfo& second) {
   if (first.is_ok && second.is_unsupported_resource &&
-      HmiInterfaces::STATE_NOT_AVAILABLE != second.interface_state) {
+      (HmiInterfaces::STATE_AVAILABLE != second.interface_state)) {
```
Comment (VVeremjova) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @pold500 Why did you change this variable?

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 25]
```diff
@@ -789,75 +811,59 @@ bool CommandRequestImpl::IsResultCodeUnsupported(
...
-      return "Social";
-    case mobile_apis::AppHMIType::BACKGROUND_PROCESS:
-      return "Background Process";
-    case mobile_apis::AppHMIType::TESTING:
-      return "Testing";
-    case mobile_apis::AppHMIType::SYSTEM:
-      return "System";
-    default:
-      return "Invalid Component Type";
+  std::string not_responding_interfaces_string = std::accumulate(
```
Comment (VVeremjova) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @pold500 please add #include <numeric>

Comment (pold500) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @VVeremjova fixed in https://github.com/smartdevicelink/sdl_core/pull/1368/commits/ab69afffe097fc66b49d474f1e18aad87f4aa6dc
